src/main/java/de/srendi/advancedperipherals/common/blocks/base/PeripheralBlockEntity.java

```diff
@@ -54,54 +48,36 @@ public PeripheralBlockEntity(BlockEntityType<?> tileEntityTypeIn, BlockPos pos,
         peripheralSettings = new CompoundTag();
     }
 
-    @NotNull
+    @Nullable
     @Override
-    public <T1> LazyOptional<T1> getCapability(@NotNull Capability<T1> cap, @Nullable Direction direction) {
-        if (cap == Capabilities.CAPABILITY_PERIPHERAL) {
-            if (peripheral == null)
-                // Perform later peripheral creation, because creating peripheral
-                // on init of tile entity cause some infinity loop, if peripheral
-                // are depend on tile entity data
-                this.peripheral = createPeripheral();
-            if (peripheral.isEnabled()) {
-                if (peripheralCap == null) {
-                    peripheralCap = LazyOptional.of(() -> peripheral);
-                } else if (!peripheralCap.isPresent()) {
-                    // Recreate peripheral to allow CC: Tweaked correctly handle
-                    // peripheral update logic, so new peripheral and old one will be
-                    // different
-                    peripheral = createPeripheral();
-                    peripheralCap = LazyOptional.of(() -> peripheral);
-                }
-                return peripheralCap.cast();
-            } else {
-                AdvancedPeripherals.debug(peripheral.getType() + " is disabled, you can enable it in the Configuration.");
-            }
-        }
-
-        if (cap == ForgeCapabilities.ITEM_HANDLER && !remove && direction != null && this instanceof IInventoryBlock) {
-            if (handler == null || !handler.isPresent())
-                handler = LazyOptional.of(() -> new SidedInvWrapper(this, Direction.NORTH));
-            return handler.cast();
+    public IPeripheral createPeripheralCap(@Nullable Direction side) {
+        if (peripheral == null)
+            // Perform later peripheral creation, because creating peripheral
+            // on init of tile entity cause some infinity loop, if peripheral
+            // are depend on tile entity data
+            this.peripheral = createPeripheral();
```

Comment:
maybe we should either both use `this.` or both not, so we can know we are accessing samething.